docs/versioning.md

```diff
@@ -0,0 +1,63 @@
+* **End-of-Support** (Phase 4) - When an SDK reaches end-of support, it will no longer receive updates or releases. Previously published releases will continue to be available via public package managers and the code will remain on GitHub. The GitHub repository may be archived. Use of an SDK which has reached end-of-support is done at the user’s discretion. We recommend users upgrade to the new major version.
+
+!!! note "Please note that the timelines shown below are illustrative and not binding"
+
+![Maintenance policy timelines](https://docs.aws.amazon.com/images/sdkref/latest/guide/images/maint-policy.png)
+
+### Dependency lifecycle
+
+Most AWS SDKs have underlying dependencies, such as language runtimes, AWS Lambda runtime, or third party libraries and frameworks. These dependencies are typically tied to the language community or the vendor who owns that particular component. Each community or vendor publishes their own end-of-support schedule for their product.
+
+The following terms are used to classify underlying third party dependencies:
+
+* [**AWS Lambda Runtime**](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html): Examples include `nodejs20.x`, `python3.12`, etc.
+* **Language Runtime**: Examples include Python 3.12, NodeJS 20, Java 17, .NET Core, etc.
+* **Third party Library**: Examples include Pydantic, AWS X-Ray SDK, AWS Encryption SDK, Middy.js, etc.
+
+Powertools for AWS Lambda follows the [AWS Lambda Runtime deprecation policy cycle](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtime-support-policy), when it comes to Language Runtime. This means we will stop supporting their respective deprecated Language Runtime _(e.g., `nodejs20.x`)_ without increasing the major SDK version.
+
+!!! note "AWS reserves the right to stop support for an underlying dependency without increasing the major SDK version"
+
+### Communication methods
+
+Maintenance announcements are communicated in several ways:
+
+* A pinned GitHub Request For Comments (RFC) issue indicating the campaign for the next major version. The RFC will outline the path to end-of-support, specify campaign timelines, and upgrade guidance.
+* AWS SDK documentation, such as API reference documentation, user guides, SDK product marketing pages, and GitHub readme(s) are updated to indicate the campaign timeline and provide guidance on upgrading affected applications.
+* Deprecation warnings are added to the SDKs, outlining the path to end-of-support and linking to the upgrade guide.
+
+To see the list of available major versions of Powertools for AWS Lambda and where they are in their maintenance lifecycle, see [version support matrix](#version-support-matrix)
+
+### Version support matrix
```

Comment:
Just a couple minor changes in the table:
- Let's use official name convention for consistency `Powertools for AWS Lambda (TypeScript)`
- Update GA date to 07/15/2022

Reply:
On it!

Reply:
Fixed GA date. For the naming, similar to Python, we use the `(<language>)` at the top, since that's the product umbrella. We don't have to add the `(<language>)` for every sentence as it's already lengthy. 

I've updated the info banner with the `(TypeScript)` to be triple sure there aren't confusions if the title isn't read either.